integrated_channels/integrated_channel/exporters/content_metadata.py

```diff
@@ -169,12 +176,32 @@ def _get_catalog_diff(self, enterprise_catalog, content_keys, force_retrieve_all
             enterprise_catalog,
             content_keys
         )
+        customer_catalog_uuids = [catalog.uuid for catalog in enterprise_customer_catalogs]
+        ContentMetadataItemTransmission = apps.get_model(
+            'integrated_channel',
+            'ContentMetadataItemTransmission'
+        )
+        unique_new_items_to_create = []
+
+        # We need to remove any potential create transmissions if the content already exists on the customer's instance
+        # under a different catalog
+        for index, item in enumerate(items_to_create):
+            item_created_in_other_catalog = ContentMetadataItemTransmission.objects.filter(
```

Comment:
This seems like a lot of database I/O work -- does it make more sense to simply pull the full set of audit record content_ids for the customer UUID, that are not "deleted" and then reconcile with that list?

Reply:
good point!